lib/rubocop/cop/rspec/pending_without_reason.rb

```diff
@@ -70,7 +70,12 @@ class PendingWithoutReason < Base
 
         # @!method skipped_by_example_method?(node)
         def_node_matcher :skipped_by_example_method?, <<~PATTERN
-          (send nil? ${#Examples.skipped #Examples.pending} ...)
+          (send nil? ${#Examples.skipped #Examples.pending})
+        PATTERN
+
+        # @!method skipped_by_example_method_with_block?(node)
+        def_node_matcher :skipped_by_example_method_with_block?, <<~PATTERN
+          ({block numblock} (send nil? ${#Examples.skipped #Examples.pending} ...) ...)
```

Comment:
There is a difference here.
It manifests this way:
```diff
        RSpec.describe Foo do
          skip 'does something'
          pending 'does something'
          skip 'does something' do
       +  ^^^^^^^^^^^^^^^^^^^^^ Give the reason for skip.
            1
          end
```

Should it be like that?

Reply:
If you look at [the implementation](https://github.com/rspec/rspec-core/blob/cb1c8f1b5b12193221729971fbe5b390a4be877b/lib/rspec/core/example_group.rb#L186) of skipped/pending/xit/..., you see that those set a bogus reason/pending message.

The goal of this cop is to enforce having a reason:
> Checks for pending or skipped examples without reason.

And the output clearly doesn't give any hint to why those were skipped/are pending:
```
  1) Foo does something
     # No reason given
     # ./spec/a_spec.rb:2

  2) Foo does something
     # Not yet implemented
     # ./spec/a_spec.rb:3
```

Reply:
Even though putting `pending` inside an example changes the behaviour as stated in the ticket, I'd say that a better way would be:
```ruby
pending "does something", pending: "not yet implemented"
```
Is it too verbose to your liking, as it just duplicates the default?
Or is it a case of explicit vs implicit?

Reply:
`pending "does something", reason: "not yet implemented"` would be better.

Reply:
Agree, but it's an RSpec DSL, we don't have much control over it here. pending...pending sounds like a tautology indeed.

Reply:
I apologize for my mistake.
The reason for the mistake was that we were checking with test code like this.

code
````ruby
pending 'This cop is not implemented yet. Feel free to submit a pull request.'
```

result
```
Pending: (Failures listed here are expected and do not affect your suite's status)

  1) RuboCop::Cop::Style::RedundantLineContinuations This cop is not implemented yet. Feel free to submit a pull request. <---- It looked to me like the output was coming from here.
     # Not yet implemented <---- But the correct one is here.
     # ./spec/foo_spec.rb:4
```

Reply:
I actually do find `pending "does something", pending: "not yet implemented"pending "does something", pending: "not yet implemented"` too verbose.

I don’t think I have ever used the form `pending "<string>"` for any other reason than it being a reminder that I need to implement a certain example (as described by `<string>`). Having to spell out `pending: "not yet implemented"` or similar is unnecessary.

At the least, we should allow users to configure if the `:pending` metadata should be enforced when calling `pending` without a block.

Reply:
In my experience, I usually have such code, as in issue example:

```ruby
describe 'does something' do
  pending 'implement later'
end
```

I like nesting `describe` and `context`. Reason? What should I write there? "was too lazy", "had no enough time"? Something reasonable, like "we have a blocker", is pretty rare. And can be in code comment, with links.